### PR TITLE
CI: check external links (excluding GH)

### PR DESCRIPTION
**Pull request by chalin**
- This is a first attempt at enabling full link checking, both internal paths and external links. Let's give this a test run. It might be an issue for PRs adding lots of new content with many new external references.
  Note: Adding external link checking, with the cache, takes at most 2 sec more on my local machine.
- Contributes to #1052
- Repurposed the `check:links` NPM script for full link checking
- Runs Prettier over `static`
- Saves the htmltest `refcache.json` file under `static` so that it gets deployed. That way each "edition" of the website will have it's own published refcache data file.
- Normalizes a few external links in the docs

Note: If / once this PR is merged, anyone submitting a PR will need to run `npm run test` (as indicated in the contributing notes) to update the refcache. We're not currently filing if that isn't done, but we might soon.

/cc @reyang

### CODE REVIEW

**Comment by chalin** [comment]
Links checked (internal and external) in less than 8 seconds. From the [build log](https://app.netlify.com/sites/opentelemetry/deploys/63f130ac02943400085bd82f):

```console
> npm run _check:links
...
htmltest started at 08:10:56 on public
--
| ========================================================================
| ✔✔✔ passed in 7.515506054s
| tested 333 documents
```

File: content/en/blog/2022/end-user-discussion-group-apac/index.md [hunk 1]
```diff
@@ -5,50 +5,50 @@ date: 2022-12-06
...
+As mentioned above, these Discussion Groups are monthly meetings for end users
+to discuss the challenges they're facing in their OTel implementations and to
+learn from each other, as well as to talk about the project itself and what is
+working, what isn't, and what they'd like to see.
 
-We encourage you to attend and ask questions to learn how other users have 
-implemented OpenTelemetry in their organizations, resolved common issues, and 
-more. Example topics that have been discussed include tail sampling, collector 
-scaling, and OpAMP; there really is no limit to the topics! We simply ask that 
-you be respectful and abide by the [Chatham House Rule](https://www.chathamhouse.org/about-us/chatham-house-rule#:~:text=The%20Rule%20reads%20as%20follows,other%20participant%2C%20may%20be%20revealed.).
```
Comment (chalin) [inline content/en/blog/2022/end-user-discussion-group-apac/index.md:27]: FYI, I dropped the higlight "hash" from the link.

Comment (chalin) [reply content/en/blog/2022/end-user-discussion-group-apac/index.md:27]: Probably best to review the changes to this file while ignoring whitespace differences.

**Review by cartermp — APPROVED** [review]

**Pull request merged**